packages/esm-patient-labs-app/src/lab-orders/add-lab-order/test-type-search.tsx

```diff
@@ -0,0 +1,235 @@
+import React, { useCallback, useMemo, useRef, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { Button, ButtonSkeleton, Layer, Search, SkeletonText, Tile } from '@carbon/react';
+import { ArrowRight, ShoppingCartArrowDown, ShoppingCartArrowUp } from '@carbon/react/icons';
+import { useDebounce, useLayoutType, useSession } from '@openmrs/esm-framework';
+import { useOrderBasket } from '@openmrs/esm-patient-common-lib';
+import { LabOrderBasketItem, prepLabOrderPostData } from '../api';
+import styles from './test-type-search.scss';
+import { TestType, useTestTypes } from './useTestTypes';
+import { createEmptyLabOrder } from './lab-order';
+
+export interface TestTypeSearchProps {
+  openLabForm: (searchResult: LabOrderBasketItem) => void;
+}
+
+export function TestTypeSearch({ openLabForm }: TestTypeSearchProps) {
+  const { t } = useTranslation();
+  const isTablet = useLayoutType() === 'tablet';
+  const [searchTerm, setSearchTerm] = useState('');
+  const debouncedSearchTerm = useDebounce(searchTerm);
+  const searchInputRef = useRef(null);
+
+  const focusAndClearSearchInput = () => {
+    setSearchTerm('');
+    searchInputRef.current?.focus();
+  };
+
+  const handleSearchTermChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchTerm(event.target.value ?? '');
+  };
+
+  return (
+    <div className={styles.searchPopupContainer}>
+      <ResponsiveWrapper isTablet={isTablet}>
+        <Search
+          size="lg"
+          placeholder={t('searchFieldPlaceholder', 'Search for a test type')}
+          labelText={t('searchFieldPlaceholder', 'Search for a test type')}
+          onChange={handleSearchTermChange}
+          ref={searchInputRef}
+          value={searchTerm}
+        />
+      </ResponsiveWrapper>
+      <TestTypeSearchResults
+        searchTerm={debouncedSearchTerm}
+        openOrderForm={openLabForm}
+        focusAndClearSearchInput={focusAndClearSearchInput}
+      />
+    </div>
+  );
+}
+
+function ResponsiveWrapper({ children, isTablet }: { children: React.ReactNode; isTablet: boolean }) {
```

Comment:
This might actually be a reasonable component to add to the style guide. We have a lot of logic very much like this, though this is cleaner than most of the implementations of it. (Could even extends things so the `useLayoutType()` hook happens inside this component.